### PR TITLE
Add EventHub metadata E2E testcases

### PR DESCRIPTION
**Pull request by Hazhzeng**
## Description

I should've done this earlier.
Adding a dependencies on azure-eventhub (which is a popular library our customers will use when dealing with eventhub trigger). This library will generate a proper EventHub event containing necessary metadata.
1. Add a scenario where an eventhub event is generated by azure-eventhub (async version). **test_eventhub_trigger_with_metadata**
2. Add a scenario where a batch of eventhub events are generated by azure-eventhub (sync version). **test_eventhub_multiple_with_metadata**
3. This can also help us to check the compatibility between azure-functions and azure-eventhub library.
4. Establish the **python_value** and **python_type** contract between azure_functions_worker.datumdef.Datum and azure.functions.meta.Datum.

---

### PR information

- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [X] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/endtoend/eventhub_batch_functions/get_eventhub_batch_triggered/function.json [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-      "partitionKey": "WillBePopulated",
-      "tableName": "EventHubBatchTest",
-      "connection": "AzureWebJobsStorage"
-    },
-    {
-      "type": "http",
-      "direction": "out",
-      "name": "$return"
-    }
-  ]
```
Comment (vrdmr) [inline tests/endtoend/eventhub_batch_functions/get_eventhub_batch_triggered/function.json:26]: Sorry - is the change only in the tab spaces?

File: tests/endtoend/eventhub_batch_functions/get_metadata_batch_triggered/function.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "scriptFile": "__init__.py",
+  "bindings": [
+    {
+      "type": "httpTrigger",
+      "direction": "in",
+      "name": "req"
+    },
+    {
+      "type": "blob",
```
Comment (vrdmr) [inline tests/endtoend/eventhub_batch_functions/get_metadata_batch_triggered/function.json:10]: Quick question - shouldn't the type be `eventHub`? Could you please describe the test a little in the pythondoc comments in the dunder init of that function?

Comment (vrdmr) [reply tests/endtoend/eventhub_batch_functions/get_metadata_batch_triggered/function.json:10]: Even a one-liner would help get the context of the E2E tests.

File: tests/endtoend/eventhub_batch_functions/get_eventhub_batch_triggered/function.json (hunk 1 above)
Comment (Hazhzeng) [reply tests/endtoend/eventhub_batch_functions/get_eventhub_batch_triggered/function.json:26]: Yes, this change is tab spaces only.

File: tests/endtoend/eventhub_batch_functions/get_metadata_batch_triggered/function.json (hunk 2 above)
Comment (Hazhzeng) [reply tests/endtoend/eventhub_batch_functions/get_metadata_batch_triggered/function.json:10]: Good point, but unfortunately I cannot make comment in json file.
These tests are all following the flow of
1. metadata_output => an HTTP trigger generating EventHub Event from azure-eventhub SDK.
2. metadata_multiple => an EventHub trigger which will be executed when the newly generated Event arrives EventHub. Writing the event metadata to a blob storage.
3.  get_metadata_batch_triggered => an HTTP trigger for getting the EventHub Event metadata from the blob, responding back to the E2E testcase.

Comment (Hazhzeng) [reply tests/endtoend/eventhub_batch_functions/get_metadata_batch_triggered/function.json:10]: I will add these comments in the .py file instead.

File: tests/endtoend/test_eventhub_batch_functions.py [hunk 3]
```diff
@@ -3,10 +3,16 @@
...
+from dateutil import parser, tz
 
 from azure_functions_worker import testutils
 
 
+# This is EventHub (cardinality: many) E2E. Each testcase consists of 3 part:
+# 1. An eventhub_output_batch HTTP trigger for generating EventHub event
+# 2. An eventhub_multiple EventHub trigger for converting event into blob
+# 3. A get_eventhub_batch_triggered HTTP trigger for getting the event body
 class TestEventHubFunctions(testutils.WebHostTestCase):
```
Comment (vrdmr) [inline tests/endtoend/test_eventhub_batch_functions.py:16]: Ideally, you could use the [docstring comments](https://www.python.org/dev/peps/pep-0257/) for the comments where you explain methods and Classes. I am open to doing that in another PR.

```suggestion
class TestEventHubFunctions(testutils.WebHostTestCase):
    """Test EventHub Triggers and Bindings
    
    This is EventHub (cardinality: many) E2E. Each testcase consists of 3 part:
    1. An eventhub_output_batch HTTP trigger for generating EventHub event
    2. An eventhub_multiple EventHub trigger for converting event into blob
    3. A get_eventhub_batch_triggered HTTP trigger for getting the event body
    """
```

**Pull request merged**